dashboards-reports/server/config/config.ts

```diff
@@ -0,0 +1,89 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import {
+  CoreSetup,
+  Logger,
+  PluginInitializerContext,
+} from '../../../../src/core/server';
+import { ReportingConfigType } from './schema';
+import { get } from 'lodash';
+import { first, map } from 'rxjs/operators';
+import { createConfig$ } from './createConfig';
+
+interface Config<BaseType> {
+  get<Key1 extends keyof BaseType>(key1: Key1): BaseType[Key1];
+  get<Key1 extends keyof BaseType, Key2 extends keyof BaseType[Key1]>(
+    key1: Key1,
+    key2: Key2
+  ): BaseType[Key1][Key2];
+  get<
+    Key1 extends keyof BaseType,
+    Key2 extends keyof BaseType[Key1],
+    Key3 extends keyof BaseType[Key1][Key2]
+  >(
+    key1: Key1,
+    key2: Key2,
+    key3: Key3
+  ): BaseType[Key1][Key2][Key3];
+  get<
+    Key1 extends keyof BaseType,
+    Key2 extends keyof BaseType[Key1],
+    Key3 extends keyof BaseType[Key1][Key2],
+    Key4 extends keyof BaseType[Key1][Key2][Key3]
+  >(
+    key1: Key1,
+    key2: Key2,
+    key3: Key3,
+    key4: Key4
```

Comment:
is key3 and key4 needed/used?

Reply:
no, currently it has only 2 keys. Removing the rest.